### PR TITLE
SITL: Aircraft correct some syntax, implicit cast, make some local va…

### PR DESCRIPTION
**Pull request by khancyr**
…riable const, remove tabs and whitespace

### CODE REVIEW

**Comment by magicrub** [comment]
Looks correct to me. Only suggestion is to move all the declarations in the constructor to be done in the class def and to not set it if non-zero. This is only building for Linux class systems so the memory footprint improvement is negligible.

File: libraries/SITL/SIM_Aircraft.cpp [hunk 1]
```diff
@@ -296,30 +297,23 @@ double Aircraft::rand_normal(double mean, double stddev)
...
         {
-            x = 2.0*rand()/RAND_MAX - 1;
-            y = 2.0*rand()/RAND_MAX - 1;
-
+            x = 2.0 * rand()/RAND_MAX - 1;
+            y = 2.0 * rand()/RAND_MAX - 1;
             r = x*x + y*y;
-        }
-        while (is_zero(r) || r > 1.0);
-        {
```
Comment (khancyr) [inline libraries/SITL/SIM_Aircraft.cpp]: this was a mess, but we agree that is was a do while function ?

Comment (magicrub) [inline libraries/SITL/SIM_Aircraft.cpp]: I think so. I'm very surprised it was written like that in the first place. You changed it to implement the same code but formatted correctly, which is much better. The question is was the real code what we actually we wanted? If noyt, that's a different PR.

File: libraries/SITL/SIM_Aircraft.cpp [hunk 2]
```diff
@@ -553,37 +547,37 @@ bool Aircraft::get_mag_field_ef(float latitude_deg, float longitude_deg, float &
     bool valid_input_data = true;
 
     /* round down to nearest sampling resolution */
-    int min_lat = (int)(latitude_deg / SAMPLING_RES) * SAMPLING_RES;
-    int min_lon = (int)(longitude_deg / SAMPLING_RES) * SAMPLING_RES;
+    int32_t min_lat = static_cast<int32_t>(static_cast<int32_t>(latitude_deg / SAMPLING_RES) * SAMPLING_RES);
+    int32_t min_lon = static_cast<int32_t>(static_cast<int32_t>(longitude_deg / SAMPLING_RES) * SAMPLING_RES);
```
Comment (OXINARF) [inline libraries/SITL/SIM_Aircraft.cpp]: Why two casts? Why not allow it to do all operations in float and only in the end cast it to int32?

File: libraries/SITL/SIM_Aircraft.cpp [hunk 3]
```diff
@@ -553,37 +547,37 @@ bool Aircraft::get_mag_field_ef(float latitude_deg, float longitude_deg, float &
...
     /* limit to table bounds - required for maxima even when table spans full globe range */
     if (latitude_deg <= SAMPLING_MIN_LAT) {
-        min_lat = SAMPLING_MIN_LAT;
+        min_lat = static_cast<int32_t>(SAMPLING_MIN_LAT);
         valid_input_data = false;
     }
 
     if (latitude_deg >= SAMPLING_MAX_LAT) {
-        min_lat = (int)(latitude_deg / SAMPLING_RES) * SAMPLING_RES - SAMPLING_RES;
+        min_lat = static_cast<int32_t>(static_cast<int32_t>(latitude_deg / SAMPLING_RES) * SAMPLING_RES - SAMPLING_RES);
```
Comment (OXINARF) [inline libraries/SITL/SIM_Aircraft.cpp]: Same.

File: libraries/SITL/SIM_Aircraft.cpp [hunk 4]
```diff
@@ -553,37 +547,37 @@ bool Aircraft::get_mag_field_ef(float latitude_deg, float longitude_deg, float &
...
 
     if (longitude_deg <= SAMPLING_MIN_LON) {
-        min_lon = SAMPLING_MIN_LON;
+        min_lon = static_cast<int32_t>(SAMPLING_MIN_LON);
         valid_input_data = false;
     }
 
     if (longitude_deg >= SAMPLING_MAX_LON) {
-        min_lon = (int)(longitude_deg / SAMPLING_RES) * SAMPLING_RES - SAMPLING_RES;
+        min_lon = static_cast<int32_t>(static_cast<int32_t>(longitude_deg / SAMPLING_RES) * SAMPLING_RES - SAMPLING_RES);
```
Comment (OXINARF) [inline libraries/SITL/SIM_Aircraft.cpp]: Same.

File: libraries/SITL/SIM_Aircraft.cpp (hunk 1 above)
Comment (OXINARF) [inline libraries/SITL/SIM_Aircraft.cpp]: http://web.archive.org/web/20150906061405/http://en.literateprograms.org/Box-Muller_transform_(C)

According to the source it is correct.

**Comment by OXINARF** [comment]
This already makes a good cleanup, any improvement can be done on top. Merged, thanks!

**Pull request merged**